### PR TITLE
refactor: 💡 Updated Manage Dropdown to HDS in Targets Menu

### PR DESCRIPTION
**Pull request by CaptFalc**
Updated the manage dropdown present in the targets scope to reflect HDS design

:tickets: ✅ Closes: https://hashicorp.atlassian.net/browse/ICU-14597

## Screenshots (if appropriate)
**Before**
Screenshot 2024-07-29 at 4 17 52 PM
Screenshot 2024-07-29 at 4 18 24 PM
**After**
Screenshot 2024-07-30 at 11 02 44 AM
Screenshot 2024-07-30 at 11 03 26 AM

## How to Test
1. Setup yarn server
2. Select Targets menu and select a resource
3. Manage dropdown should appear on the right and reflect the new design.

## Checklist
- [X] I have added before and after screenshots for UI changes
- [] I have added JSON response output for API changes
- [X] I have added steps to reproduce and test for bug fixes in the description
- [] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ui/admin/tests/acceptance/targets/host-sources-test.js [hunk 1]
```diff
@@ -196,9 +199,9 @@ module('Acceptance | targets | host-sources', function (hooks) {
     await visit(urls.target);
 
     await click(`[href="${urls.targetHostSources}"]`);
-
+    await click(MANAGE_DROPDOWN_SELECTOR);
     assert
-      .dom('.rose-layout-page-actions a')
+      .dom('[data-test-manage-targets-dropdown] ul li a')
```
[the remark this answers is not in the document]
Comment (CaptFalc) [reply ui/admin/tests/acceptance/targets/host-sources-test.js:204]: Sounds good, went ahead and did it! Thanks for the heads up

File: ui/admin/app/components/targets/target/actions/index.hbs [hunk 2]
```diff
@@ -2,44 +2,40 @@
...
+
   {{#if (can 'delete model' @model)}}
-    <dropdown.separator />
-    <dropdown.button
-      @style='danger'
+    <dd.Separator />
+    <dd.Interactive
+      @text={{t 'resources.target.actions.delete'}}
+      @color='critical'
       @disabled={{@model.canSave}}
```
Comment (DhariniJeeva) [inline ui/admin/app/components/targets/target/actions/index.hbs:36]: I don't think we have a disabled attribute, `dd.interactive` only allows HTML disabled attributes(without the `@`)

Comment (CaptFalc) [reply ui/admin/app/components/targets/target/actions/index.hbs]: Thanks Dharini for the heads up, went ahead and made those changes 😄

**Review by ZedLi — APPROVED** [review]
Looks great!

**Review by cameronperera — APPROVED** [review]
Looks good!

**Pull request merged**